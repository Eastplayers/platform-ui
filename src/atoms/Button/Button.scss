@import "../../styles/abstracts/main";

.pui-btn {
  --btn-primary: var(--primary);
  --btn-primary-dark: var(--primary-dark);

  --btn-bg: var(--btn-primary);
  --btn-text: var(--white);
  --btn-hover-text: var(--white);
  --btn-border: var(--btn-primary);
  --btn-hover-bg: var(--btn-primary-dark);

  border: none;
  height: pxToRem(40);
  padding: 0 pxToRem(12);
  font-weight: 600;
  font-size: pxToRem(14);
  line-height: pxToRem(17);
  border-radius: pxToRem(8);
  cursor: pointer;
  transition: all 0.25s ease-in-out;
  will-change: background;
  outline: none;
  background: var(--btn-bg);
  color: var(--btn-text);
  border: 1.5px solid var(--btn-border);

  &:hover {
    background: var(--btn-hover-bg);
    color: var(--btn-hover-text);
  }

  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }

  &.pui-btn-lg {
    height: pxToRem(45);
    font-size: pxToRem(16);
    line-height: pxToRem(19);
  }

  &.pui-btn-outline {
    --btn-bg: transparent;
    --btn-text: var(--btn-primary);
    --btn-hover-bg: var(--btn-primary);
    --btn-hover-text: var(--white);
  }

  &.pui-btn-borderless {
    --btn-bg: transparent;
    --btn-text: var(--btn-primary);
    --btn-hover-bg: rgba(23, 24, 26, 0.08);
    --btn-hover-text: var(--btn-primary-dark);
    --btn-border: transparent;
  }

  &.pui-btn-secondary {
    --btn-primary: var(--grey-900);
    --btn-primary-dark: var(--grey-900);
  }

  &.pui-btn-danger {
    --btn-primary: var(--accent-red);
    --btn-primary-dark: var(--accent-red);
  }
}
