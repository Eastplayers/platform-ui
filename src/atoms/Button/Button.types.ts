import { ButtonHTMLAttributes, DetailedHTMLProps, ReactNode } from "react";

export interface ButtonProps
  extends DetailedHTMLProps<
    ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > {
  children?: ReactNode;
  size?: ButtonSizes;
  variant?: ButtonVariants;
  disabled?: boolean;
}

export enum ButtonSizes {
  BIG = "big",
  MEDIUM = "medium",
}

export enum ButtonVariants {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  GHOST = "ghost",
}
