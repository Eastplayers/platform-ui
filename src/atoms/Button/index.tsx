export * from "./Button.types";
export { default } from "./Button";
