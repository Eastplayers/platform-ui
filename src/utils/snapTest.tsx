import renderer from "react-test-renderer";
import React from "react";

export interface Config {
  props: object;
  description: string;
}

const singleSnapTest = (tree: any, description: string) => {
  it(description, () => {
    expect(tree).toMatchSnapshot();
  });
};

const testSnapshots = (Component: any, configs: Config[]) => {
  describe("snapshots", () => {
    configs.forEach((config) => {
      const { props, description } = config;

      const tree = renderer.create(<Component {...props} />).toJSON();
      singleSnapTest(tree, description);
    });
  });
};

export { testSnapshots };
