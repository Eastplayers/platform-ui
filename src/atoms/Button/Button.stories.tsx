import React from "react";
import Button from "./Button";

export const SingleStory = (): React.ReactElement => {
  return <Button>Hello</Button>;
};

export default {
  title: "Atoms/Button",
};
