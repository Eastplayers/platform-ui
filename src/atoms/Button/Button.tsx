import classNames from "classnames";
import React, { FC } from "react";
import "./Button.scss";
import { ButtonProps, ButtonSizes, ButtonVariants } from "./Button.types";

const Button: FC<ButtonProps> = ({ children, variant, size, ...rest }) => {
  return (
    <button
      className={classNames("pui-btn", {
        [`pui-btn-${variant}`]: !!variant,
        [`pui-btn-${size}`]: !!size,
      })}
      {...rest}
    >
      {children}
    </button>
  );
};

Button.defaultProps = {
  variant: ButtonVariants.PRIMARY,
  size: ButtonSizes.MEDIUM,
};

export default Button;
