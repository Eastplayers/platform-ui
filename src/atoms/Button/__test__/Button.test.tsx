import { testSnapshots } from "../../../utils/snapTest";
import Button from "../Button";
import renderer from "react-test-renderer";
import React from "react";
import { render, screen } from "@testing-library/react";
import { userEvent } from "@storybook/testing-library";
import "@testing-library/jest-dom";

// const breadcrumb = [
//   {
//     label: "Home",
//     path: "/",
//     hightLightLabel: "Home",
//   },
// ];

describe("Button snapshot", () => {
  testSnapshots(Button, [
    {
      props: {},
      description: "Should render correctly",
    },
  ]);
});

describe("Button logic", () => {
  test("Trigger onClick fc when click Button", () => {
    render(<Button children="Click me" />);
    const button = screen.getByText("Click me");
    // await userEvent.click(button);
    expect(button).toHaveTextContent("Click me");
  });
});
