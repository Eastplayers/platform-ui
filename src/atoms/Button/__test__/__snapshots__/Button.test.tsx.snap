// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="pui-btn pui-btn-primary pui-btn-medium"
/>
`;

exports[`Button logic snapshots Should render correctly 1`] = `
<button
  className="pui-btn pui-btn-primary pui-btn-medium"
/>
`;

exports[`Button snapshot snapshots Should render correctly 1`] = `
<button
  className="pui-btn pui-btn-primary pui-btn-medium"
/>
`;
